### PR TITLE
Iterator To Array functionality

### PR DESCRIPTION
**Pull request by andriy-invoiced**
Added iterator to array functionality

### CODE REVIEW

File: src/Iterator.php [hunk 1]
```diff
@@ -241,4 +241,13 @@ private function rangeStart(int $pointer, int $limit)
     {
         return floor($pointer / $limit) * $limit;
     }
+
+    /**
+     * Cast Iterator to array
+     * @return array
```
Comment (jaredk2g) [inline src/Iterator.php]: This phpdoc is unnecessary

File: tests/IteratorTest.php [hunk 2]
```diff
@@ -99,8 +99,9 @@ public function testNext()
 
     public function testRewind()
     {
+        self::$iterator->next();
```
Comment (jaredk2g) [inline tests/IteratorTest.php:102]: Can you explain the change behind this test case?

**Review by jaredk2g — CHANGES_REQUESTED** [review]

File: tests/IteratorTest.php (hunk 2 above)
Comment (andriy-invoiced) [reply tests/IteratorTest.php:102]: in your initial test case iterator may already be ate the first position. So, this change will make sure, you are running rewinding function on array being iterated.

File: tests/IteratorTest.php [hunk 3]
```diff
@@ -227,6 +228,19 @@ public function testOffsetGet()
         $this->assertEquals(1, self::$iterator[1]->id());
     }
 
+    public function testToArray()
+    {
+        $query = new Query(IteratorTestModel::class);
+        $query->limit(3);
+        $iterator = new Iterator($query);
+        $iterator->next();
```
Comment (jaredk2g) [inline tests/IteratorTest.php]: Is it a requirement for next() to be called first in order for toArray() to work?

Comment (andriy-invoiced) [reply tests/IteratorTest.php]: no, we just check that array order is not broken, because of iteration

File: tests/IteratorTest.php (hunk 2 above)
Comment (jaredk2g) [reply tests/IteratorTest.php:102]: Ok that makes sense

**Comment by jaredk2g** [comment]
Great change. LGTM

**Pull request merged**